### PR TITLE
Pure component notes

### LINKED ISSUE #277: Hot Module Reloading not working with functional components
Hot Module Reloading is not yet working with functional components, according to  https://github.com/gaearon/babel-plugin-react-transform/issues/57

Took me some time to get to that link. Just putting this out here in case someone else has the same issue with `AboutView` not hot reloading.

Looks like until it's fixed, it might be better to not use functional components. HMR gives me so much more than functional components at this point.

Also, this is a great starter kit. Exactly what I wanted without all the universal fuzz, making it so much harder to learn. Thank you for the great work! :star2:

### PR DESCRIPTION
**Pull request by justingreenberg**
resolves #277

### CODE REVIEW

File: tests/layouts/CoreLayout.spec.js [hunk 1]
```diff
@@ -34,11 +34,4 @@ describe('(Layout) Core', function () {
     expect(_component.type).to.equal('div')
   })
 
-  it('Should render a child component.', function () {
```
Comment (davezuko) [inline tests/layouts/CoreLayout.spec.js]: This is because the test utils don't work with pure components yet, right?

Comment (justingreenberg) [inline tests/layouts/CoreLayout.spec.js]: exactly... do you want me to revert this commit to leave this as component class?

Comment (davezuko) [inline tests/layouts/CoreLayout.spec.js]: Not at all, was just making sure. And I'm not totally caught up to speed here: A regular React component class rendered inside of a pure component will still work w/ HMR, right?

Comment (justingreenberg) [inline tests/layouts/CoreLayout.spec.js]: @davezuko yes all transforms are working, see below as usual. i also tested hmr with multiple nested `AboutView`s to be sure react-transform is walking the full component tree :+1: 

hmr-stateless

Comment (justingreenberg) [inline tests/layouts/CoreLayout.spec.js]: on a separate note, do you want me to make a nice demo screencast for readme?

Comment (davezuko) [inline tests/layouts/CoreLayout.spec.js]: That would be _amazing_, and thanks for all the gifs, they help a lot!

**Pull request merged**